### PR TITLE
Aqui se actualiza la base de datos y se checa si existe

### PR DESCRIPTION
**Pull request by IzuOvando**
Falta subir cambios a prisma y poner a que hora se actualice cuando se sube un nuevo archivo, ya se agregan dropdowns extra y se empezo a crear una tabla de visualizacion.

Gracias los amo

### CODE REVIEW

File: pages/api/checkFile.tsx
Comment (Ruben35) [inline pages/api/checkFile.tsx:1]: This should be implemented with app router api routes

File: src/app/api/files/upload.tsx
Comment (Ruben35) [inline src/app/api/files/upload.tsx:1]: This won't work cause is not a route handler for app/api. It has specific conventions

Please see: [https://nextjs.org/docs/app/building-your-application/routing/route-handlers](https://nextjs.org/docs/app/building-your-application/routing/route-handlers)

File: src/app/page.tsx [hunk 1]
```diff
@@ -1,10 +1,68 @@
...
 import { Sheet, SheetTrigger } from "@/components/ui/sheet";
 import { PrintersAside } from "@/components";
-export default async function Home() {
+import FilterDropdown from "@/components/filters/FilterDropdown";
+import FileUpload from "@/components/files/FileUpload";
+
+export default function Home() {
+  const [shouldShowButtons, setShouldShowButtons] = useState(false);
+  const isServer = typeof window === 'undefined';
+  const baseUrl = isServer ? 'http://localhost:3000' : '';
```
Comment (Ruben35) [inline src/app/page.tsx:12]: why not just use and env variable?

File: src/app/page.tsx [hunk 2]
```diff
@@ -1,10 +1,68 @@
...
 import { Button } from "@/components/ui/button";
 import { Sheet, SheetTrigger } from "@/components/ui/sheet";
 import { PrintersAside } from "@/components";
-export default async function Home() {
+import FilterDropdown from "@/components/filters/FilterDropdown";
+import FileUpload from "@/components/files/FileUpload";
+
+export default function Home() {
+  const [shouldShowButtons, setShouldShowButtons] = useState(false);
+  const isServer = typeof window === 'undefined';
```
Comment (Ruben35) [inline src/app/page.tsx:11]: this component is client cause the use of "use client"

File: src/app/page.tsx [hunk 3]
```diff
@@ -1,10 +1,68 @@
...
+  const baseUrl = isServer ? 'http://localhost:3000' : '';
+
+  const checkFile = async () => {
+    try {
+      const response = await fetch(`${baseUrl}/api/checkFile`, {
```
Comment (Ruben35) [inline src/app/page.tsx:16]: This endpoint could be replaced with an action and not use any states, just await the return of the action to check the existance of the file and then use the returned boolean to show the button or not

File: src/components/Navbar.tsx [hunk 4]
```diff
@@ -1,3 +1,4 @@
+"use server";
```
Comment (Ruben35) [inline src/components/Navbar.tsx:1]: no need, by default all components are server

File: src/components/files/FileUpload.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+        console.log("entro")
+        const file = event.target.files ? event.target.files[0] : null;
+        if (!file) return;
+
+        const formData = new FormData();
+        formData.append('file', file);
+
+        try {
+            console.log("1")
+            const response = await fetch('http://localhost:3000/src/app/api/files/upload', {
```
Comment (Ruben35) [inline src/components/files/FileUpload.tsx:20]: use env

Comment (Ruben35) [reply src/components/files/FileUpload.tsx:20]: wrong api route

File: src/components/files/FileUpload.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    const triggerFileInput = () => {
+        fileInputRef.current?.click();
+    };
+
+    return (
+        <div>
+            <Input
+                ref={fileInputRef}
+                type="file"
+                onChange={handleFileUpload}
```
Comment (Ruben35) [inline src/components/files/FileUpload.tsx:58]: why the upload is on the OnChange of the input and not triggered directly by the button?

File: src/components/filters/FilterDropdown.tsx [hunk 7]
```diff
@@ -0,0 +1,85 @@
+"use client";
+import React, { useEffect, useState } from 'react';
+import DropdownButton from '@/components/ui/dropdown';
+import { PrismaClient } from '@prisma/client';
```
Comment (Ruben35) [inline src/components/filters/FilterDropdown.tsx:4]: Please, use instead, this is the client as a singleton for a proper use on Next.js
```suggestion
import prisma from "@/lib/db";
```

File: src/components/filters/FilterDropdown.tsx [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+import TableViewer from './TableView'
+import getFilteredVouchers from './GetFilters'
+
+interface FilterResults {
+    name: string;
+}
+
+const prisma = new PrismaClient();
```
Comment (Ruben35) [inline src/components/filters/FilterDropdown.tsx:12]: This don't work on a client component

File: src/components/filters/FilterDropdown.tsx [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+const FilterDropdown: React.FC = () => {
+    const [filters, setFilters] = useState<Filters>({ empresa: [], material: [] });
+    const [selectedFilters, setSelectedFilters] = useState<SelectedFilters>({ empresa: 'Ninguno', material: 'Ninguno' });
+    const [tableData, setTableData] = useState<any[]>([]);
+
+    useEffect(() => {
+        const loadFilters = async () => {
+            try {
+                const filterQueries = {
+                    empresa: prisma.$queryRaw<FilterResults[]>`SELECT DISTINCT empresa AS name FROM User WHERE empresa <> '' AND empresa IS NOT NULL`,
```
Comment (Ruben35) [inline src/components/filters/FilterDropdown.tsx:33]: Why do this querys raw, prisma methods have them auto

File: src/components/filters/GetFilters.tsx [hunk 10]
```diff
@@ -0,0 +1,12 @@
+import { PrismaClient, Prisma } from '@prisma/client';
+
+const prisma = new PrismaClient();
+
+async function getFilteredVouchers(filters: Prisma.UserWhereInput) {
```
Comment (Ruben35) [inline src/components/filters/GetFilters.tsx:5]: this is not a componen, should not be here, should be more as action

File: src/components/filters/GetFilters.tsx [hunk 11]
```diff
@@ -0,0 +1,12 @@
+import { PrismaClient, Prisma } from '@prisma/client';
```
Comment (Ruben35) [inline src/components/filters/GetFilters.tsx:1]: The same component of the import of prisma client

File: src/components/ui/dropdown.tsx [hunk 12]
```diff
@@ -0,0 +1,42 @@
+"use client";
+import React, { useState } from 'react';
```
Comment (Ruben35) [inline src/components/ui/dropdown.tsx:2]: Why not use the shadcn component?

**Review by Ruben35 — COMMENTED** [review]
there are many comments about the impl

File: src/components/ui/dropdown.tsx (hunk 12 above)
Comment (IzuOvando) [reply src/components/ui/dropdown.tsx:2]: solve

File: src/components/filters/FilterDropdown.tsx (hunk 9 above)
Comment (IzuOvando) [reply src/components/filters/FilterDropdown.tsx:33]: solve

File: src/components/filters/FilterDropdown.tsx (hunk 8 above)
Comment (IzuOvando) [reply src/components/filters/FilterDropdown.tsx:12]: solve

File: prisma/schema.prisma [hunk 13]
```diff
@@ -14,3 +14,24 @@ model User {
...
+  operador String
+  checador String
+  createdAt DateTime @default(now())
+}
+
+model QueryView {
+  id       Int      @id @default(autoincrement())
+  filterName     String
+  filters  String     
+  userName   String
```
Comment (Ruben35) [inline prisma/schema.prisma:35]: username is not needed, or it is?
also if it is, should be all lowercase()

Comment (IzuOvando) [reply prisma/schema.prisma:35]: I thought you could search the saved filters by username, to filter them, right?

File: src/components/filters/GetFilters.tsx (hunk 10 above)
Comment (IzuOvando) [reply src/components/filters/GetFilters.tsx:5]: import { PrismaClient, Prisma } from '@prisma/client';

const prisma = new PrismaClient();
const { TicketWhereInput } = Prisma;

async function getFilteredVouchers(filters: TicketWhereInput) {
  const filteredVouchers = await prisma.ticket.findMany({
    where: filters,
  });
  return filteredVouchers;
}


This is what you mean, I didn't quite understand the comment.

File: src/components/filters/GetFilters.tsx (hunk 11 above)
Comment (IzuOvando) [reply src/components/filters/GetFilters.tsx:1]: import { PrismaClient, Prisma } from '@prisma/client';

const prisma = new PrismaClient();
const { TicketWhereInput } = Prisma;

async function getFilteredVouchers(filters: TicketWhereInput) {
const filteredVouchers = await prisma.ticket.findMany({
where: filters,
});
return filteredVouchers;
}

This is what you mean, I didn't quite understand the comment.

File: src/components/filters/GetFilters.tsx (hunk 10 above)
Comment (IzuOvando) [reply src/components/filters/GetFilters.tsx:5]: already move to actions

**Review by Ruben35 — APPROVED** [review]
Perfect ✨ Well done 💯

**Pull request merged**